src/content_mapblock.cpp

```diff
@@ -1513,7 +1513,49 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
 
 			v3f pos = intToFloat(p, BS);
 
-			std::vector<aabb3f> boxes = n.getNodeBoxes(nodedef);
+			u8 neighbors = 0;
+
+			// locate possible neighboring nodes to connect to
+			if (f.node_box.type == NODEBOX_CONNECTED) {
+				v3s16 p2 = p;
+
+				p2.Y++;
+				MapNode n2 = data->m_vmanip.getNodeNoEx(blockpos_nodes + p2);
```

Comment:
Additionally, I used //fixedpos to place a 10x2000 long carpet of these into an area that wasn't mapgen'd to make sure it would be crossing map boundaries at mapgen time, and after travelling to the other end of the 2000 node long carpet, I found that

* there were no breakages in the drawn connections
* mapgen had cut caves through the carpet
* there was snow on some parts of the carpet

BTW, if this would be broken then classic fences would break to, since this uses the same method.

There could be a problem with mapgen placing these nodes, but that's so far-fetched I'm not concerned.